### ISSUE TITLE
feat(forge): cheatcode for getting the current prank

### ISSUE THREAD

**Issue by PaulRBerg** [foundry-rs/foundry#4875]
### Component

Forge

### Describe the feature you would like

It would be helpful to have a cheatcode for reading the current prank. The use case is testing utilities that don't have knowledge of when the current prank was started, but which require tweaking the `msg.sender` to perform a particular action in the contracts.

Suggested names:

- `getPrank`
- `currentPrank`
- `readPrank`

**Comment by xeno097** [comment]
If no one is working on this I'd be happy to give this a try. Any tip or suggestion on how to get this done would be more than welcome

**Comment by Evalir** [comment]
Hey @xeno097, awesome. For adding a cheatcode, [this article](https://dev.to/teddav/foundry-add-a-cheatcode-5hc8) can be really useful. Be mindful that after adding a cheatcode we usually ask for a follow up PR for forge-std and the foundry book (i'll point this out again on the PR)

Before you're fully off to the races, I'd like to loop in @mds1 to see if he has any opinions about this cheatcode's design

**Comment by xeno097** [comment]
Considering that the `prank` cheat code allows impersonating the 0 address I think that the function signature should be something like this: 

```solidity
function readPrank() external returns (bool, address, address)
```

The first `bool` would indicate if a prank is active, the first `address` would be the new `msg.sender`, and the last one the new `tx.origin` used to set the prank. This way if both addresses are 0, the boolean would indicate if there is an active prank or if the user is actually impersonating the 0 address.

**Comment by Evalir** [comment]
Closing—fixed by #4884

**Comment by PaulRBerg** [comment]
wicked
